src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/JsonUtils.cs

```diff
@@ -165,5 +165,26 @@ public static void WriteBehaviourStateToStringBuilder(StringBuilder stringBuilde
                 }
             },
         };
+
+        public static readonly JsonSerializerSettings JsonSerializerSettings = new()
+        {
+            Formatting = Formatting.None,
+            ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
+            ContractResolver = JsonConverterContractResolver.Instance,
+            Error = delegate(object _, ErrorEventArgs args)
+            {
+                // just eat certain errors
+                if (args.ErrorContext.Error is MissingComponentException || args.ErrorContext.Error.InnerException is UnityException or NotSupportedException or MissingComponentException)
```

Comment:
What is the implication of eating NotSupportedException? Is this for specific settings of serialization or actually reading/writing?

Reply:
These are here and have been for a while (Not new to this PR), just moved/renamed code.

They handle converting unity Objects to json, specifically/normally around monobehaviours or other objects we don't have explicit converters for.  When reading all fields of game objects and behaviours in unity, many of them throw exceptions trying to access the field as they aren't meant to be read in user space, or not on the render thread, or other unity specific conditions.

This block prevents those conditions from stopping the other fields on the object being written